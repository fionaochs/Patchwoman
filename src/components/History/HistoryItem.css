.HistoryItem {
  h3 {
    text-transform: uppercase;
    color: white;
    margin: 0 auto;
  }

  p {
    font-weight: 400;
    font-size: 0.8rem;
  }
}