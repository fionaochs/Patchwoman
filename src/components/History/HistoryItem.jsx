import React from 'react';
import PropTypes from 'prop-types';
import styles from './HistoryItem.css';

const HistoryItem = ({ method, url, onClick }) => {
  return (
    <section className={styles.HistoryItem} onClick={onClick}>
      <h3>{method}</h3>
      <p>{url}</p>
    </section>
  );
};

HistoryItem.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export default HistoryItem;
