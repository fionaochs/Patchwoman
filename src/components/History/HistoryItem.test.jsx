import React from 'react';
import { shallow } from 'enzyme';
import HistoryItem from './HistoryItem.jsx';

describe('RequestItem', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <HistoryItem method='post' url='https://jsonplaceholder.typicode.com/posts' onClick={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
