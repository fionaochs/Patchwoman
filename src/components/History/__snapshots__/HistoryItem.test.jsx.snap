// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestItem matches a snapshot 1`] = `
<section
  className="HistoryItem"
  onClick={[Function]}
>
  <h3>
    post
  </h3>
  <p>
    https://jsonplaceholder.typicode.com/posts
  </p>
</section>
`;
